### PR TITLE
install_fips: Create the OPENSSLDIR as it might not exist

### LINKED ISSUE #16564: Openssl 3.0: install_fips in cross-compiling fails to install fips config file
Judging from the code, it should have at least another line to make sure the folder where the fips config file is going to end up is created/exists

```
install_fips: build_sw $(INSTALL_FIPSMODULECONF)
	@[ -n "$(INSTALLTOP)" ] || (echo INSTALLTOP should not be empty; exit 1)
	@$(PERL) $(SRCDIR)/util/mkdir-p.pl $(DESTDIR)$(MODULESDIR)
	@$(ECHO) "*** Installing FIPS module"
	@$(ECHO) "install $(INSTALL_FIPSMODULE) -> $(DESTDIR)$(MODULESDIR)/$(FIPSMODULENAME)"
	@cp "$(INSTALL_FIPSMODULE)" $(DESTDIR)$(MODULESDIR)/$(FIPSMODULENAME).new
	@chmod 755 $(DESTDIR)$(MODULESDIR)/$(FIPSMODULENAME).new
	@mv -f $(DESTDIR)$(MODULESDIR)/$(FIPSMODULENAME).new \
	       $(DESTDIR)$(MODULESDIR)/$(FIPSMODULENAME)
	@$(ECHO) "*** Installing FIPS module configuration"
	@$(ECHO) "install $(INSTALL_FIPSMODULECONF) -> $(DESTDIR)$(OPENSSLDIR)/fipsmodule.cnf"
	@cp $(INSTALL_FIPSMODULECONF) $(DESTDIR)$(OPENSSLDIR)/fipsmodule.cnf
```

I'd expect to find a line that does this before installing the config

```
@$(PERL) $(SRCDIR)/util/mkdir-p.pl $(DESTDIR)$(OPENSSLDIR)
```

Else I get a build error on the install_fips target that it can't copy the fips config file to that location

```
install providers/fipsmodule.cnf -> /Volumes/devel/qnx_repos/stage/nto/aarch64le/usr/ssl/fipsmodule.cnf
install apps/openssl -> /Volumes/devel/qnx_repos/stage/nto/aarch64le/usr/bin/openssl
cp: /Volumes/devel/qnx_repos/stage/nto/aarch64le/usr/ssl/fipsmodule.cnf: No such file or directory
```

### PR DESCRIPTION
**Pull request by t8m**
Fixes #16564

### CODE REVIEW

**Comment by t8m** [comment]
@levitte I suppose VMS does not need this as I do not see any explicit creation of the MODULESDIR either?

**Review by paulidale — APPROVED** [review]
Does VMS need this too?

**Comment by levitte** [comment]
Why doesn't `install_fips` simply depend on `install_ssldirs`?

**Comment by levitte** [comment]
> @levitte I suppose VMS does not need this as I do not see any explicit creation of the MODULESDIR either?

https://github.com/openssl/openssl/blob/5ecf10a0d2fb1c858b25afd5e48eafe6ef76edd4/Configurations/descrip.mms.tmpl#L626-L633

**Comment by t8m** [comment]
> Why doesn't `install_fips` simply depend on `install_ssldirs`?

Good point.

**Comment by t8m** [comment]
Hmm, actually not. I think install_ssldirs is doing way too much. The install_fips should just install the fips module and fipsmodule.conf and not mess with other things.

**Comment by t8m** [comment]
OK, mkdir for VMS added as fixup. @paulidale please re-review.

File: Configurations/descrip.mms.tmpl [hunk 1]
```diff
@@ -569,6 +569,8 @@ uninstall_docs : uninstall_html_docs
 {- output_off() if $disabled{fips}; "" -}
 install_fips : build_sw $(INSTALL_FIPSMODULECONF)
 	@ WRITE SYS$OUTPUT "*** Installing FIPS module"
+        - CREATE/DIR ossl_installroot:[MODULES{- $sover_dirname.$target{pointer_size} -}.'arch']
+        - CREATE/DIR/PROT=(S:RWED,O:RWE,G:RE,W:RE) OSSL_DATAROOT:[000000]
```
Comment (levitte) [inline Configurations/descrip.mms.tmpl:573]: Curious alignment...

**Review by paulidale — APPROVED** [review]
Once the alignment thingy is worked out.

File: Configurations/descrip.mms.tmpl (hunk 1 above)
Comment (t8m) [reply Configurations/descrip.mms.tmpl]: For some reason tabs are expanded to just one space in this file on github diff view. Fixup pushed

**Comment by t8m** [comment]
> Once the alignment thingy is worked out.

Fixup pushed

**Review by levitte — APPROVED** [review]

**Review by paulidale — APPROVED** [review]

**Comment by t8m** [comment]
Merged to master and 3.0 branches. Thank you for the reviews.

**Pull request closed without merging**